pkg/network/kuryr_bootstrap.go

```diff
@@ -0,0 +1,802 @@
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find worker nodes subnet")
+	}
+	log.Printf("Found worker nodes subnet %s", workerSubnetId)
+	routerId, err := findOpenStackRouterId(client, "openshift-external-router", tag)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find worker nodes router")
+	}
+	log.Printf("Found worker nodes router %s", routerId)
+	ps, err := getOpenStackRouterPorts(client, routerId)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed list ports of worker nodes router")
+	}
+
+	if !lookupOpenStackPort(ps, svcSubnetId) {
+		log.Printf("Ensuring service subnet router port with %s IP", ipStr)
+		portId, err := ensureOpenStackPort(client, "kuryr-service-subnet-router-port", tag,
+			svcNetId, svcSubnetId, ipStr)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create service subnet router port")
+		}
+		log.Printf("Service subnet router port %s present, adding it as interface.", portId)
+		err = ensureOpenStackRouterInterface(client, routerId, nil, &portId)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create service subnet router interface")
+		}
+	}
+
+	masterSgId, err := findOpenStackSgId(client, "master", tag)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find master nodes security group")
+	}
+	log.Printf("Found master nodes security group %s", masterSgId)
+	workerSgId, err := findOpenStackSgId(client, "worker", tag)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find worker nodes security group")
+	}
+	log.Printf("Found worker nodes security group %s", workerSgId)
+
+	log.Print("Ensuring pods security group")
+	podSgId, err := ensureOpenStackSg(client, "kuryr-pods-security-group", tag)
+	log.Printf("Pods security group %s present", podSgId)
+
+	log.Print("Allowing traffic from masters and nodes to pods")
+	err = ensureOpenStackSgRule(client, podSgId, masterSgId)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to add rule opening traffic from masters")
+	}
+	err = ensureOpenStackSgRule(client, podSgId, workerSgId)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to add rule opening traffic from workers")
+	}
+	err = ensureOpenStackSgRule(client, masterSgId, podSgId)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to add rule opening traffic to masters")
+	}
+	err = ensureOpenStackSgRule(client, workerSgId, podSgId)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to add rule opening traffic to workers")
+	}
+	log.Print("All requried traffic allowed")
+
+	lbClient, err := openstack.NewLoadBalancerV2(provider, gophercloud.EndpointOpts{})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create Octavia client")
+	}
+
+	// We need first usable IP from services CIDR
+	// This will get us the first one (subnet IP)
+	ip = iputil.FirstUsableIP(*svcNet)
+	ipStr = ip.String()
+	log.Printf("Creating OpenShift API loadbalancer with IP %s", ipStr)
```

Comment:
Does the next means we are creating our own lb in front of the master, regardless of the one created by the installer? If that so, this will only be used by the kuryr components, right?

Reply:
I believe we talked this through on IRC, but let's repeat it here for the record. Currently the LB created by installer is a service VM. They plan to move to Octavia, but there's no ETA yet and they will still need to support deployments without Octavia. It makes sense to make Kuryr responsible for that LB.